src/lib_ccx/ccx_decoders_xds.c

```diff
@@ -667,12 +667,13 @@ int xds_do_current_and_future (struct cc_subtitle *sub, struct ccx_decoders_xds_
 		case XDS_TYPE_ASPECT_RATIO_INFO:
 			{
 				unsigned ar_start, ar_end;
-				int changed = 0;
+				// int changed = 0; /* Currently unused */
 				was_proc = 1;
 				if (ctx->cur_xds_payload_length < 5) // We need 2 data bytes
 					break;
-				if (!ctx->cur_xds_payload[2] & 20 || !ctx->cur_xds_payload[3] & 20) // Bit 6 must be 1
-					break;
+				if (~ctx->cur_xds_payload[2] & 0x20 || ~ctx->cur_xds_payload[3] & 0x20) // Bit 6 must be 1 
+					break; // Should it be inverted? 
+
```

Comment:
This was correct as it was. If the bits 6 of those 2 bytes are not 1 then data is invalid so we skip it.

Reply:
But how do you check it with (int) 20? You use unary operators with int that is in binary 10100. To check for 6th bit, mustn't you use 0x20?

I got about ! operator being right, it just makes little sense (as says compiler) to use operators on bits with it.

Reply:
Ah yes - I didn't see the 0x20. Instead I see that sign instead of the !. Please leave ! because that is correct.
But yes it's 0x20 indeed.